### PR TITLE
Rewrite tests and add linting

### PR DESCRIPTION
**Pull request by paksu**
- Replace `chai` and `mocha` with `jest`
- Add `xo` to lint
- Rewrite tests in jest
- Lint tests

Deliberately don't touch production code yet. This just touches the tests.

### CODE REVIEW

**Comment by idmitme** [comment]
Looks good to me, approved. Only one thing confused me: in some tests you used `beforeEach` and in others `beforeAll`. I noticed that `beforeEach` is used only where there is only one `it` check, so I guess `beforeEach` is used to guard from undesired side effects in case if we add more `it` statements,  and beforeAll is used to speed up the tests, but still not sure what was your reasoning behind this.

File: test/test.js [hunk 1]
```diff
@@ -162,6 +162,43 @@ describe('mongoose-auto-increment plugin', () => {
...
+      const Project = mongoose.model('Project', projectSchema);
+
+      project1 = await new Project({ name: 'Cool project', apple: { name: 'Apple' }, google: { name: 'Google' } }).save();
+      project2 = await new Project({ name: 'Another Cool project' }).save();
+      project2.google = { name: 'Google' };
+      project2 = await project2.save();
+    });
+
+
+    it('increases counter when subdocument is created with the main document', async () => {
```
Comment (idmitme) [inline test/test.js]: Unnecessary `async`

**Comment by idmitme** [comment]
Continue our discussion how to make these tests better. If you want we can add a few more tests for corner cases (there are plenty of them). For example we don't check that we create unique index by default, we only check that it's not unique if { unique: false } option provided, which is clearly not enough. We do not check behaviour when field is present already but it has not a number type, we don't check what happens when we are close to maxExplicitValue.

**Comment by paksu** [comment]
I'll try to see if I can make every setup be a `beforeEach`. The problem that I had with it was that I'd need a way to reset the database between test runs
